spec/apollo-federation/tracing_spec.rb

```diff
@@ -4,307 +4,320 @@
+          end
         end
-      end
 
-      query_obj = Class.new(GraphQL::Schema::Object) do
-        graphql_name 'Query'
+        query_obj = Class.new(GraphQL::Schema::Object) do
+          graphql_name 'Query'
 
-        field :items, [item_obj], null: false
+          field :items, [item_obj], null: false
 
-        def items
-          [{ id: '1' }, { id: '2' }]
+          def items
+            [{ id: '1' }, { id: '2' }]
+          end
+        end
+
+        Class.new(base_schema) do
+          query query_obj
         end
       end
 
-      Class.new(base_schema) do
-        query query_obj
+      let(:trace) do
+        result = schema.execute('{ items { id, name } }', context: { tracing_enabled: true })
+        described_class.attach_trace_to_result(result)
+        ApolloFederation::Tracing::Trace.decode(Base64.decode64(result[:extensions][:ftv1]))
       end
-    end
 
-    let(:trace) do
-      result = schema.execute('{ items { id, name } }', context: { tracing_enabled: true })
-      described_class.attach_trace_to_result(result)
-      ApolloFederation::Tracing::Trace.decode(Base64.decode64(result[:extensions][:ftv1]))
+      it 'records index instead of response_name for items in arrays' do
+        expected_trace = ApolloFederation::Tracing::Trace.new(
+          start_time: { seconds: 1_564_920_001, nanos: 0 },
+          end_time: { seconds: 1_564_920_002, nanos: 0 },
+          duration_ns: 11,
+          root: {
+            child: [{
+              response_name: 'items',
+              type: '[Item!]!',
+              start_time: 1,
+              end_time: 2,
+              parent_type: 'Query',
+              child: [
+                {
+                  index: 0,
+                  child: [{
+                    response_name: 'id',
+                    type: 'String!',
+                    start_time: 3,
+                    end_time: 4,
+                    parent_type: 'Item',
+                  }, {
+                    response_name: 'name',
+                    type: 'String!',
+                    start_time: 5,
+                    end_time: 6,
+                    parent_type: 'Item',
+                  },],
+                },
+                {
+                  index: 1,
+                  child: [{
+                    response_name: 'id',
+                    type: 'String!',
+                    start_time: 7,
+                    end_time: 8,
+                    parent_type: 'Item',
+                  }, {
+                    response_name: 'name',
+                    type: 'String!',
+                    start_time: 9,
+                    end_time: 10,
+                    parent_type: 'Item',
+                    error: [{
+                      message: "Can't continue with this query",
+                      location: [{ line: 1, column: 15 }],
+                      json: %({"message":"Can't continue with this query","locations":[{"line":1,"column":15}],"path":["items",1,"name"]}),
```

Comment:
I removed some whitespace here. Previously it had some whitespace added into it.